modules/consensus/synchronize.go

```diff
@@ -417,17 +417,19 @@ func (cs *ConsensusSet) threadedRPCRelayHeader(conn modules.PeerConn) error {
 		return cs.validateHeader(boltTxWrapper{tx}, h)
 	})
 	cs.mu.RUnlock()
-	if err == errOrphan {
-		// If the header is an orphan, try to find the parents. Call needs to
-		// be made in a separate goroutine as execution requires calling an
-		// exported gateway method - threadedRPCRelayHeader was likely called
-		// from an exported gateway function.
-		//
-		// NOTE: In general this is bad design. Rather than recycling other
-		// calls, the whole protocol should have been kept in a single RPC.
-		// Because it is not, we have to do weird threading to prevent
-		// deadlocks, and we also have to be concerned every time the code in
-		// managedReceiveBlocks is adjusted.
+	// WARNING Orphan Multithreading Logic (dangerous area)
```

Comment:
needs a colon: `WARNING:`

I think though we typically use `WARN:` in the codebase, so it should probably just be the 4 letters and a colon.